### DISCUSSION TITLE
Nuxt 3 cannot retrieve dynamic param on API GET function

### DISCUSSION THREAD

**Discussion by JMillCode** [nuxt/nuxt#20938] [Questions] [answered]
Hello, I need your help with Nuxt 3. I work with Nuxt since few years and i freshly start Nuxt 3 and I don't get it. Nuxt 3.4.2 with firebase 9.21.0 (Vue 3.2) I followed firebase instructions so I set my app like this and I can't pass a param to my function =>

**./lib/firebase.js =>**

const app = initializeApp(firebaseConfig);
export const firestoreDb = getFirestore(app);

**./server/api/getProduct.js**

import { firestoreDb } from "../../lib/firebase";
import { getDocs, collection, doc, getDoc } from 'firebase/firestore'

export default defineEventHandler(async (event) => {
    try {
     const docRef = doc(firestoreDb, "cities", ???**param** ???);
     const docSnap = await getDoc(docRef);     
      return docSnap.data();
    } catch(error) {
      return error
   }
})
**./pages/[id].vue**

export default {
  async setup() {
    const product = await useFetch(`/api/getProduct`, { params: { city: 'Paris'} });
  }
}
But I don't understand how can I retrive my param (route.params.id) There is something I don't understand because when I try to import { firestoreDb } from "../../lib/firebase"; from a plugins or a store, I get a nuxt error telling my that I can't call those lib outside the server file. I need some inlightment or doc link to understand how make a GET call with a param from a page to the server.

I try to make a GET call with param and I don't understant how to retrieve my param from route or what ever solution to make a call to my DB. I can see in event.context.originalUrl: '/api/getProduct?cyty=Paris', but i feel like isn't right to get it from there. I try readBody(event) but I got "Error: (405 HTTP method is not allowed. (/api/getProduct?city=Paris

Thank you for any help.

**Comment by maou-shonen** [comment, marked as answer]
`param` refers to the part of the variable in the path,
Your example is using `getQuery(event).city`

**Reply by JMillCode** [reply]
Thank you for your help it works ! But why their is no reference of getQuery in Nuxt documentation or in JS documentaion ? I am confused about this solution. 
Thank you anyway

**Reply by maou-shonen** [reply]
Same as `~/utils`,
`~/server/utils` will auto import
https://nuxt.com/docs/guide/directory-structure/server
![image](https://github.com/nuxt/nuxt/assets/22576780/33d8c5e2-042a-4878-92b4-81e32fc3e7d6)

**Reply by JMillCode** [reply]
Thank you so much. 👏
